packages/dev/core/src/Rendering/IBLShadows/iblShadowsSpatialBlurPass.ts

```diff
@@ -106,67 +114,105 @@ export class _IblShadowsSpatialBlurPass {
+        this._update();
+
+        let counter = 0;
+        this._scene.onBeforeRenderObservable.add(() => {
+            counter = 0;
+        });
+        this._scene.onAfterRenderTargetsRenderObservable.add(() => {
+            if (++counter == 2) {
+                if (this.enabled && this._outputTexture.isReady()) {
+                    this._update();
+                    this._outputTexture.render();
+                }
+            }
         });
     }
 
-    private _updatePostProcess(effect: Effect) {
+    private _update() {
+        this._outputTexture.setTexture("voxelTracingSampler", this._renderPipeline.getVoxelTracingTexture());
         const iterationCount = 1;
-        effect.setVector4("blurParameters", new Vector4(iterationCount, this._worldScale, 0.0, 0.0));
-        const prePassRenderer = this._scene.prePassRenderer;
-        if (prePassRenderer) {
-            const wnormalIndex = prePassRenderer.getIndex(Constants.PREPASS_WORLD_NORMAL_TEXTURE_TYPE);
-            const depthIndex = prePassRenderer.getIndex(Constants.PREPASS_DEPTH_TEXTURE_TYPE);
-            if (wnormalIndex >= 0) effect.setTexture("worldNormalSampler", prePassRenderer.getRenderTarget().textures[wnormalIndex]);
-            if (depthIndex >= 0) effect.setTexture("linearDepthSampler", prePassRenderer.getRenderTarget().textures[depthIndex]);
+        this._outputTexture.setVector4("blurParameters", new Vector4(iterationCount, this._worldScale, 0.0, 0.0));
+        const geometryBufferRenderer = this._scene.geometryBufferRenderer;
+        if (!geometryBufferRenderer) {
+            return;
         }
+        const depthIndex = geometryBufferRenderer.getTextureIndex(GeometryBufferRenderer.SCREENSPACE_DEPTH_TEXTURE_TYPE);
+        this._outputTexture.setTexture("depthSampler", geometryBufferRenderer.getGBuffer().textures[depthIndex]);
+        const wnormalIndex = geometryBufferRenderer.getTextureIndex(GeometryBufferRenderer.NORMAL_TEXTURE_TYPE);
+        this._outputTexture.setTexture("worldNormalSampler", geometryBufferRenderer.getGBuffer().textures[wnormalIndex]);
+    }
+
+    /**
+     * Called by render pipeline when canvas resized.
+     * @param scaleFactor The factor by which to scale the canvas size.
+     */
+    public resize(scaleFactor: number = 1.0) {
+        const newSize = {
+            width: Math.max(1.0, Math.floor(this._engine.getRenderWidth() * scaleFactor)),
+            height: Math.max(1.0, Math.floor(this._engine.getRenderHeight() * scaleFactor)),
+        };
+        this._outputTexture.resize(newSize, false);
     }
 
     /**
      * Checks if the pass is ready
      * @returns true if the pass is ready
      */
     public isReady() {
-        return this._outputPP.isReady() && !(this._debugPassPP && !this._debugPassPP.isReady());
+        return this._outputTexture && this._outputTexture.isReady() && !(this._debugPassPP && !this._debugPassPP.isReady());

```

Comment:
`_outputTexture` is created in the constructor, so can't be undefined/null.

Reply:
Done